spec/beef/extensions/websocket_hooked_browser_spec.rb

```diff
@@ -0,0 +1,106 @@
+# encoding: UTF-8
+require 'rest-client'
+require 'core/main/network_stack/websocket/websocket'
+require 'websocket-client-simple'
+
+RSpec.describe 'BeEF Extension WebSockets' do
+
+  before(:all) do
+    @config = BeEF::Core::Configuration.instance
+    @cert_key = @config.get('beef.http.https.key')
+    @cert = @config.get('beef.http.https.cert')
+    @port = @config.get('beef.http.websocket.port')
+    @secure_port = @config.get('beef.http.websocket.secure_port')
+    @config.set('beef.http.websocket.secure', true)
+    @config.set('beef.http.websocket.enable', true)
```

Comment:
As this is not the default in the config.yaml (```beef.http.websocket.enable = true```), you'll need to change it back in an after(:all) block or this change will persist after this test is complete, potentially unknowingly altering the users config. e.g.

```
after(:all) do
  @config.set('beef.http.websocket.enable', false)
end
```

Your process killing on lines 97/101 should go there as well rather than inside the test block. That is the conventional RSpec way to handle clean-up for spec tests (like with the set-up in the before block).

Reply:
Nice catch!
I didn't use after all specifically because of an issue it was creating with closing an http session if i wanted to include it( in the comment). I didn't think there was much of a point if there was only going to be the local + pid/s. I am going to add it now that there is more though.